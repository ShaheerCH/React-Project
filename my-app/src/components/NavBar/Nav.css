.navBar {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  justify-items: center;
  align-items: center;
  box-sizing: border-box;
  background-color: white;
  position: fixed;
  top: 0;
  z-index: 3;
  border-bottom: 1px solid #e8e8e8;
  padding: 10px 50px;
  box-shadow: rgba(0, 0, 0, 0.15) 1.95px 1.95px 2.6px;
}

.logo-container {
  display: flex;
  align-items: center;
}

h1 {
  margin-left: 15px;
  font-family: 'Poppins', sans-serif;
  font-size: 25px;
}

.navUl {
  margin: 0%;
  padding: 0%;
  display: flex;
  flex-wrap: wrap;
}

.nav-link {
  margin: 0 15px;
  font-family: 'Poppins', sans-serif;
  font-size: 12px;
  font-weight: 500;
  letter-spacing: 1.9px;
  text-decoration: none;
  color: #2088ff;
}

.nav-link.active {
  text-decoration: underline;
  color: #0056b4;
}

.navUl > li:last-of-type {
  border-left: 2px solid #000;
  padding-left: 20px;
  margin-left: 10px;
  box-sizing: border-box;
}

@media screen and (max-width: 980px) {
  .navBar {
    padding: 0 30px;
  }

  h1 {
    font-size: 18px;
  }

  .logo-img {
    width: 40px;
    height: 40px;
  }

  .nav-link {
    font-size: 10px;
    margin: 0 10px;
  }

  .navUl > li:last-of-type {
    padding-left: 5px;
  }
}
