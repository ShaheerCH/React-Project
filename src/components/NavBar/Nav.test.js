import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Nav from './Nav';

describe('Jest test', () => {
  it('Match Snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Nav />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
