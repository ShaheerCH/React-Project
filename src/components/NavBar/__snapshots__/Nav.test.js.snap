// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest test Match Snapshot 1`] = `
<nav
  class="navBar"
>
  <div
    class="logo-container"
  >
    <img
      alt="Cool logo."
      class="logo-img"
      height="60"
      src="planet.png"
      width="60"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    class="navUl"
  >
    <li>
      <a
        class="nav-link"
        href="/rockets"
      >
        <span>
          ROCKETS
        </span>
      </a>
    </li>
    <li>
      <a
        class="nav-link"
        href="/missions"
      >
        <span>
          MISSIONS
        </span>
      </a>
    </li>
    <li>
      <a
        class="nav-link"
        href="/profile"
      >
        <span>
          MY PROFILE
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
